### PR TITLE
Refactor system mounts to be placed on the config

### PR DESCRIPTION
**Pull request by crosbymichael**
Also remove the RestrictSys bool replaced by configurable paths that the
user can specify.

Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

### CODE REVIEW

File: integration/template_test.go [hunk 1]
```diff
@@ -49,9 +51,42 @@ func newTemplateConfig(rootfs string) *configs.Config {
...
+				Flags:       defaultMountFlags,
+			},
+			{
+				Source:      "mqueue",
+				Destination: "/dev/mqueue",
+				Device:      "mqueue",
+				Flags:       defaultMountFlags,
+			},
+			{
+				Source:      "devpts",
```
Comment (mrunalp) [inline integration/template_test.go]: I am thinking maybe this should also be done by default. The ptmx setup depends on it.

Comment (crosbymichael) [inline integration/template_test.go]: good point, i'll change it

**Comment by crosbymichael** [comment]
@mrunalp moved pts back to defaults

**Comment by mrunalp** [comment]
LGTM. Let's see what others think :)

**Comment by crosbymichael** [comment]
ping @LK4D4

**Comment by dqminh** [comment]
LGTM

**Comment by avagin** [comment]
LGTM

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**